docs/src/browsers.md

```diff
@@ -686,14 +630,199 @@ $env:PLAYWRIGHT_DOWNLOAD_HOST="192.0.2.1"
+```
+
+```powershell tab=bash-powershell lang=csharp
+$env:PLAYWRIGHT_BROWSERS_PATH="$env:USERPROFILE\pw-browsers"
+pwsh bin/Debug/netX/playwright.ps1 install
+```
+
+When running Playwright scripts, ask it to search for browsers in a shared location.
+
+```bash tab=bash-bash lang=js
+PLAYWRIGHT_BROWSERS_PATH=$HOME/pw-browsers npx playwright test
+```
 
-## Skip browser downloads
+```batch tab=bash-batch lang=js
+set PLAYWRIGHT_BROWSERS_PATH=%USERPROFILE%\pw-browsers
+npx playwright test
+```
+
+```powershell tab=bash-powershell lang=js
+$env:PLAYWRIGHT_BROWSERS_PATH="$env:USERPROFILE\pw-browsers"
+npx playwright test
+```
+
+```bash tab=bash-bash lang=python
+# Linux/macOS
+PLAYWRIGHT_BROWSERS_PATH=$HOME/pw-browsers python playwright_script.py
+```
+
+```batch tab=bash-batch lang=python
+set PLAYWRIGHT_BROWSERS_PATH=%USERPROFILE%\pw-browsers
+python playwright_script.py
+```
+
+```powershell tab=bash-powershell lang=python
+
+$env:PLAYWRIGHT_BROWSERS_PATH="$env:USERPROFILE\pw-browsers"
+python playwright_script.py
+```
+
+```bash tab=bash-bash lang=java
+PLAYWRIGHT_BROWSERS_PATH=$HOME/pw-browsers mvn test
+```
+
+```batch tab=bash-batch lang=java
+set PLAYWRIGHT_BROWSERS_PATH=%USERPROFILE%\pw-browsers
+mvn test
+```
+
+```powershell tab=bash-powershell lang=java
+$env:PLAYWRIGHT_BROWSERS_PATH="$env:USERPROFILE\pw-browsers"
+mvn test
+```
+
+```bash tab=bash-bash lang=csharp
+PLAYWRIGHT_BROWSERS_PATH=$HOME/pw-browsers dotnet test
+```
+
+```batch tab=bash-batch lang=csharp
+set PLAYWRIGHT_BROWSERS_PATH=%USERPROFILE%\pw-browsers
+dotnet test
+```
+
+```powershell tab=bash-powershell lang=csharp
+$env:PLAYWRIGHT_BROWSERS_PATH="$env:USERPROFILE\pw-browsers"
+dotnet test
+```
+
+Playwright keeps track of packages that need those browsers and will garbage collect them as you update Playwright to the newer versions.
+
+:::note
+Developers can opt-in in this mode via exporting `PLAYWRIGHT_BROWSERS_PATH=$HOME/pw-browsers` in their `.bashrc`.
+:::
+
+### Hermetic install
+* langs: js
+
+You can opt into the hermetic install and place binaries in the local folder:
+
+
+```bash tab=bash-bash
+# Places binaries to node_modules/playwright-core/.local-browsers
+PLAYWRIGHT_BROWSERS_PATH=0 npx playwright install
+```
+
+```batch tab=bash-batch
+# Places binaries to node_modules\playwright-core\.local-browsers
+set PLAYWRIGHT_BROWSERS_PATH=0
+npx playwright install
+```
+
+```powershell tab=bash-powershell
+# Places binaries to node_modules\playwright-core\.local-browsers
+$env:PLAYWRIGHT_BROWSERS_PATH=0
+npx playwright install
+```
+
+:::note
+`PLAYWRIGHT_BROWSERS_PATH` does not change installation path for Google Chrome and Microsoft Edge.
+:::
+
+### Skip browser downloads
 
 In certain cases, it is desired to avoid browser downloads altogether because
 browser binaries are managed separately.
 
 This can be done by setting `PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD` variable before installation.
 
+```bash tab=bash-bash lang=js
```

Comment:
strictly speaking we need 9 more code snippets here, but I'd vote for not adding language snippets, since this is a very niece use-case.